addons/stock_account/models/account_invoice.py

```diff
@@ -77,6 +77,56 @@ def _anglo_saxon_sale_move_lines(self, i_line):
                 ]
         return []
 
+    def _get_related_stock_moves(self):
+        """ To be overridden for customer invoices and vendor bills in order to
+        return the stock moves related to this invoice.
+        """
+        return self.env['stock.move']
+
+    def _get_products_set(self):
+        """ Returns a recordset of the products contained in this invoice's lines
+        """
+        return self.mapped('invoice_line_ids.product_id')
+
+    def _get_anglosaxon_interim_account(self, product):
+        """ Returns the interim account used in anglosaxon accounting for
+        this invoice"""
+        if self.type in ('out_invoice', 'out_refund'):
+            return product.product_tmpl_id._get_product_accounts()['stock_output']
+        elif self.type in ('in_invoice', 'in_refund'):
+            return product.product_tmpl_id.get_product_accounts()['stock_input']
+
+        return None
+
+    def invoice_validate(self):
```

Comment:
This is another issue that need to be addressed here, while having many lines and the lines needs to be created 
in mass 
https://github.com/odoo/odoo/blob/master/addons/account/models/account_invoice.py#L1220
the `check_move_validity` needs to be passed in context to avoids wasting too much time while creating lines `assert_balanced` method is called three times when is possible to have it called once at Journal Entry posting.


https://github.com/Vauxoo/addons-vauxoo/blob/8.0/account_anglo_saxon_stock_move/model/invoice.py#L64

Reply:
not related to the task, and already taken care in another branch

Reply:
Hi @qdp-odoo could please address me to that branch
> not related to the task, and already taken care in another branch

Regards.

Reply:
https://github.com/odoo/odoo/pull/21498